images/dptp-3312-hypershift-leaks-cleaner/cleanup-hypershift-leaks.sh

```diff
@@ -1,19 +1,37 @@
 #!/bin/bash
 NOW=$(date +%s)
 echo Cleanning up resources older than $(date -u -d @$NOW +%Y-%m-%dT%H:%M:%SZ)
-AWS_CRED_PATH=${AWS_CRED_PATH:-/token/.awscred}
-echo Using AWS_CRED_PATH=${AWS_CRED_PATH}
+
+# int his script we are assuming that the credentials are mounted according to their cluster-profile
+# e.g. cluster-profile=aws, AWS_CONFIG_FILE=/aws/token/.awscred
+# e.g. cluster-profile=aws2, AWS_CONFIG_FILE=/aws2/token/.awscred
+updateCredentialsEnv() {
+    cluster_profile=`oc -n clusters get hostedclusters/$1 -o json | jq -r '.metadata.annotatios."cluster-profile"'`
+    if [ $cluster_profile == null ]; then 
+        echo "error: No cluster profile found for $1."
+        return
+    fi
+    if ! [ -f "$cluster_profile/token" ]; then
+        echo "error: Cluster profile secret: $cluster_profile not present."
+        return
+    fi
+    AWS_CONFIG_FILE=${AWS_CONFIG_FILE:-/$cluster_profile/token/.awscred}
+}
+
 
 echo 'Deleting broken HostedClusters...'
 # Get json output from oc command
-oc -n clusters get hostedclusters -o json | jq -r '.items[] | select(.metadata.annotations.broken == "true" and .spec.dns.baseDomain == "hypershift.aws-2.ci.openshift.org" ) | "\(.metadata.namespace) \(.metadata.name)"' | while read cluster; do
+oc -n clusters get hostedclusters -o json | jq -r '.items[] | select(.metadata.annotations.broken == "true" ) | " \(.metadata.name)"' | while read cluster; do
     echo 'Deleting' $cluster
-    echo $cluster | awk -v AWS_CRED_PATH="$AWS_CRED_PATH" '{ print "timeout 5m hypershift destroy cluster aws --name " $2 " --aws-creds " AWS_CRED_PATH " --destroy-cloud-resources " }' | bash
+    updateCredentialsEnv $cluster
+    if [ $cluster_profile != null ]; then 
+        echo $cluster | awk -v AWS_CONFIG_FILE="$AWS_CONFIG_FILE" '{ print "timeout 5m hypershift destroy cluster aws --name " $cluster " --aws-creds " AWS_CONFIG_FILE " --destroy-cloud-resources " }' | bash
```

Comment:
we probably can improve this line to not using `awk`, just something like
`timeout 5m hypershift destroy cluster aws --name "$cluster" --aws-creds "$AWS_CONFIG_FILE" --destroy-cloud-resources" may work?